### PR TITLE
Rating and limits preso

### PR DESCRIPTION
**Pull request by jamesaimonetti**
KazooCon '15 presentation source material, plus some code cleanup

### CODE REVIEW

File: applications/jonny5/src/j5_authz_req.erl [hunk 1]
```diff
@@ -121,24 +124,28 @@ maybe_reseller_limited(Request) ->
...
-            end
+            check_reseller_limits(Request, ResellerId)
     end.
 
--spec maybe_authorize(j5_request:request(), j5_limits:limits()) -> j5_request:request().
+-spec check_reseller_limits(j5_request:request(), ne_binary()) -> 'ok'.
+check_reseller_limits(Request, ResellerId) ->
+    Limits = j5_limits:get(ResellerId),
+    R = maybe_authorize(Request, Limits),
+    (not j5_request:is_authorized(R, Limits))
```
Comment (fenollp) [inline applications/jonny5/src/j5_authz_req.erl]: Braces superfluous

File: applications/jonny5/src/j5_flat_rate.erl [hunk 2]
```diff
@@ -61,9 +64,13 @@ eligible_for_flat_rate(Request) ->
...
     TrunkBlacklist = ?BLACKLIST,
-    (wh_util:is_empty(TrunkWhitelist) orelse re:run(Number, TrunkWhitelist) =/= nomatch)
+    (wh_util:is_empty(TrunkWhitelist)
+     orelse re:run(Number, TrunkWhitelist) =/= 'nomatch'
+    )
         andalso
-          (wh_util:is_empty(TrunkBlacklist) orelse re:run(Number, TrunkBlacklist) =:= nomatch).
+          (wh_util:is_empty(TrunkBlacklist)
+           orelse re:run(Number, TrunkBlacklist) =:= 'nomatch'
+          ).
```
Comment (fenollp) [inline applications/jonny5/src/j5_flat_rate.erl]: Braaaaaceeeeees

File: applications/jonny5/src/j5_authz_req.erl (hunk 1 above)
Comment (jamesaimonetti) [inline applications/jonny5/src/j5_authz_req.erl]: You're superfluous.

File: applications/jonny5/src/j5_flat_rate.erl (hunk 2 above)
Comment (jamesaimonetti) [inline applications/jonny5/src/j5_flat_rate.erl]: for clarityyyyyy

File: applications/jonny5/src/j5_authz_req.erl (hunk 1 above)
Comment (siplabs) [inline applications/jonny5/src/j5_authz_req.erl]: Ha-ha!

**Pull request merged**